apis/v1alpha1/httproute_types.go

```diff
@@ -532,6 +546,45 @@ type HTTPRequestMirrorFilter struct {
 	Port PortNumber `json:"port"`
 }
 
+// HTTPRequestAuthorization defines configuration for the RequestAuthorization filter.
+type HTTPRequestAuthorization struct {
+	// URL indicates the address of the service used to perform authorization checks.
+	//
+	// There are two protocols supported:
+	// - grpc: In this mode, the authorization server should receive a grpc request of type <service.auth.v3.CheckRequest>.
+	//   It is expected for the authorization server to implement the Envoy ExtAuthz GRPC protocol:
+	//   https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto
+	//
+	// - https: An HTTP request that will include all the Headers and Request Path, but no body content.
+	//   It is expected for the authorization server to implement the Envoy ExtAuthz HTTP protocol:
```

Comment:
Not all HTTP external auth mechanisms are compatible with the envoy way. There's TLS and Host header considerations to take into account as well.